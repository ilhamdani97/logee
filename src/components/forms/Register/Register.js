import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Text from '../../fields/Text_V2';
import Password from '../../fields/Password';
import styles from './styles.scoped.css';
import { Button, Tooltip } from 'logeeport-ui';
import { useSelector } from 'react-redux';

const Register = (props) => {
  const { handleSubmit, valid } = props;
  const nameInputProps = {
    placeholder: 'Contoh: Muhammad Fahmi Faisal Hikmawan',
  };
  const npwpInputProps = { placeholder: 'Contoh: 123456789000000' };
  const phoneInputProps = { placeholder: 'Contoh: 081234567890' };
  const textInputProps = { placeholder: 'Contoh: email@logeeport.id' };
  const passwordInputProps = {
    placeholder: 'Min. 8 karakter terdiri dari huruf besar & angka',
  };

  const { message, isLoading } = useSelector((s) => s.register);

  return (
    <>
      <form className={styles.root}>
        <div className={styles.info}>
          <Field
            component={Text}
            inputProps={npwpInputProps}
            label="Nomor NPWP Perusahaan"
            name="npwp"
          />
          <div className={styles.infoForm}>
            <Tooltip content={<p>Pastikan nomor NPWP Benar, <br />digunakan untuk aktivasi akun <br />dan verifikasi transaksi.</p>} direction="bottom">
              <img src="/assets/ic-info-circle.svg" />
            </Tooltip>
          </div>
        </div>
        <div className={styles.info}>
          <Field
            component={Text}
            inputProps={phoneInputProps}
            label="Nomor Handphone Anda"
            name="phone"
          />
          <div className={styles.infoForm}>
            <Tooltip content={<p>Pastikan nomor Anda Aktif <br />dan Benar, digunakan untuk <br />menerima SMS OTP transaksi.</p>} direction="bottom">
              <img src="/assets/ic-info-circle.svg" />
            </Tooltip>
          </div>
        </div>
        <Field
          component={Text}
          inputProps={textInputProps}
          label="Email Anda"
          name="email"
        />
        <Field
          component={Text}
          inputProps={nameInputProps}
          label="Nama Lengkap Anda"
          name="name"
        />
        <Field
          className={styles.password}
          component={Password}
          inputProps={passwordInputProps}
          label="Kata Sandi Akun Anda"
          name="password"
        />
        <Field
          component={Password}
          inputProps={passwordInputProps}
          label="Ulangi Kata Sandi Akun Anda"
          name="retypepassword"
        />
        <Button
          disabled={isLoading.fetchRegister || !valid}
          isLoading={isLoading.fetchRegister}
          onClick={handleSubmit}
        >
          Daftar
        </Button>
        <div className={styles.term}>
          Dengan mendaftar, kamu telah menyetujui <span>Kebijakan Privasi</span>
        </div>
      </form>

      {message.fetchRegister && <span className={styles.error}>{message.fetchRegister}</span>}
    </>
  );
};

Register.defaultProps = {
  handleSubmit: () => {},
  valid: true,
};

Register.propTypes = {
  handleSubmit: PropTypes.func,
  valid: PropTypes.bool,
};

export default Register;
