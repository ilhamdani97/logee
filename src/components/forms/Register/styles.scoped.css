.root {
  margin-top: 20px;
}
.root > button {
  margin-top: 20px;
  width: 100%;
}

.error {
  margin-top: 0.5rem;
  color: var(--error-color);
}

.password {
  margin-bottom: 12px;
}

.term {
  margin-top: 20px;
}

.term span {
  text-align: center;
  color: var(--primary-color);
  font-weight: 700;
}

.info {
  position: relative;
}

.infoForm {
  position: absolute;
  top: 0;
  right: 0;
}

.infoForm div div {
  top: 20px;
  background-color: black;
  height: 77px;
  text-align: center;
  padding: 10px 10px;
}
