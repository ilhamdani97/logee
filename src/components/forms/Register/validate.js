import { formatEmail } from '@utils/format';

export default function validate(values) {
  const { email, password, phone, name, npwp, retypepassword } = values;

  return {
    email: !email && 'required' || !formatEmail(email) && 'Format email salah',
    password: !password && 'required' || password.length < 6 && 'minimal 6 karakter',
    phone: !phone && 'required',
    name: !name && 'required',
    npwp: !npwp && 'required',
    retypepassword: !retypepassword && 'required' || retypepassword !== password && 'password tidak match'
  };
}
