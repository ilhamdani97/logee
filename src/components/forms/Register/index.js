import { reduxForm } from 'redux-form';
import Component from './Register';
import validate from './validate';

export default reduxForm({
  form: 'register',
  initialValues: {
    email:'',
    password:''
  },
  validate,
})(Component);
