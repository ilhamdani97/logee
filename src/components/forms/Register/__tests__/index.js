import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Register from '../Register';
import { useSelector } from 'react-redux';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    isLoading:false,
    message: 'test'
  };
});

describe('src/components/forms/Register', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Register />);
    expect(tree).toMatchSnapshot();
  });
});
