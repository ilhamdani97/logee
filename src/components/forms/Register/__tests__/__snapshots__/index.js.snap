// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/forms/Register render 1`] = `
<React.Fragment>
  <form>
    <div>
      <Field
        component={[Function]}
        inputProps={
          Object {
            "placeholder": "Contoh: 123456789000000",
          }
        }
        label="Nomor NPWP Perusahaan"
        name="npwp"
      />
      <div>
        <Tooltip
          content={
            <p>
              Pastikan nomor NPWP Benar, 
              <br />
              digunakan untuk aktivasi akun 
              <br />
              dan verifikasi transaksi.
            </p>
          }
          delay={0}
          direction="bottom"
        >
          <img
            src="/assets/ic-info-circle.svg"
          />
        </Tooltip>
      </div>
    </div>
    <div>
      <Field
        component={[Function]}
        inputProps={
          Object {
            "placeholder": "Contoh: 081234567890",
          }
        }
        label="Nomor Handphone Anda"
        name="phone"
      />
      <div>
        <Tooltip
          content={
            <p>
              Pastikan nomor Anda Aktif 
              <br />
              dan Benar, digunakan untuk 
              <br />
              menerima SMS OTP transaksi.
            </p>
          }
          delay={0}
          direction="bottom"
        >
          <img
            src="/assets/ic-info-circle.svg"
          />
        </Tooltip>
      </div>
    </div>
    <Field
      component={[Function]}
      inputProps={
        Object {
          "placeholder": "Contoh: email@logeeport.id",
        }
      }
      label="Email Anda"
      name="email"
    />
    <Field
      component={[Function]}
      inputProps={
        Object {
          "placeholder": "Contoh: Muhammad Fahmi Faisal Hikmawan",
        }
      }
      label="Nama Lengkap Anda"
      name="name"
    />
    <Field
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": [Function],
          "type": [Function],
        }
      }
      inputProps={
        Object {
          "placeholder": "Min. 8 karakter terdiri dari huruf besar & angka",
        }
      }
      label="Kata Sandi Akun Anda"
      name="password"
    />
    <Field
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": [Function],
          "type": [Function],
        }
      }
      inputProps={
        Object {
          "placeholder": "Min. 8 karakter terdiri dari huruf besar & angka",
        }
      }
      label="Ulangi Kata Sandi Akun Anda"
      name="retypepassword"
    />
    <Button
      buttonProps={Object {}}
      className=""
      color="default"
      disabled={false}
      fullwidth={false}
      isLoading={false}
      loadingType="text"
      onClick={[Function]}
      size="medium"
    >
      Daftar
    </Button>
    <div>
      Dengan mendaftar, kamu telah menyetujui 
      <span>
        Kebijakan Privasi
      </span>
    </div>
  </form>
</React.Fragment>
`;
